sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -474,14 +475,24 @@ void ng_ipv6_netif_init_by_dev(void)
                            sizeof(if_type)) != -ENOTSUP) &&
             (if_type == NG_NETTYPE_SIXLOWPAN)) {
             uint16_t src_len = 8;
-            DEBUG("Set 6LoWPAN flag\n");
+            uint16_t max_frag_size = UINT16_MAX;
+
+            DEBUG("ipv6 netif: Set 6LoWPAN flag\n");
             ipv6_ifs[i].flags |= NG_IPV6_NETIF_FLAGS_SIXLOWPAN;
             /* use EUI-64 (8-byte address) for IID generation and for sending
              * packets */
             ng_netapi_set(ifs[i], NETCONF_OPT_SRC_LEN, 0, &src_len,
                           sizeof(src_len)); /* don't care for result */
-        }
 
+            if (ng_netapi_get(ifs[i], NETCONF_OPT_MAX_PACKET_SIZE,
+                              0, &max_frag_size, sizeof(max_frag_size)) < 0) {
+                /* if error we assume it works */
```

Comment:
> Well the IPv6 layer usually knows very well about the (in case of 6LoWPAN perceived) MTU of the device

How does it do so?

If this value is only used for 6LoWPAN fragmentation, I would vote for either exiting here or setting this value to a fixed size.